authzed/api/v1/permission_service.proto

```diff
@@ -457,6 +463,37 @@ message LookupSubjectsRequest {
 
   // context consists of named values that are injected into the caveat evaluation context
   google.protobuf.Struct context = 6 [ (validate.rules).message.required = false ];
+
+  // optional_concrete_limit, if non-zero, specifies the limit on the number of
+  // *concrete* (non-wildcard) subjects to return before the stream is closed on the
+  // server side. By default, the stream will continue resolving concrete subjects
+  // until exhausted or the stream is closed due to the client or a network issue.
+  //
+  // NOTE: Wildcard subjects ("*") have special treatment when cursors and limits are used. Because
+  // wildcards can apply to *any* concrete subjects, if a wildcard subject is found within the dataset,
+  // a wildcard subject can be returned for *all* LookupSubjects calls, regardless of the cursor or
+  // limit.
+  //
+  // For example, if wildcards are requested, a wildcard subject exists, there is a specified limit
+  // of 10 concrete subjects, and at least 10 concrete subjects exist, the API will return 11 subjects
+  // in total: the 10 concrete + the wildcard
+  //
+  // Furthermore, if a wildcard has a set of exclusions generated by the dataset,
+  // the exclusions *will respect the cursor* and only a *partial* set of exclusions will be returned
+  // for each invocation of the API.
+  //
+  // ***IT IS UP TO THE CALLER IN THIS CASE TO COMBINE THE EXCLUSIONS IF DESIRED***
+  uint32 optional_concrete_limit = 7 [(validate.rules).uint32 = {gte:0, lte: 1000}];
+
+  // optional_cursor, if specified, indicates the cursor after which results should resume being returned.
+  // The cursor can be found on the LookupSubjectsResponse object.
+  //
+  // NOTE: See above for notes about how cursors interact with wildcard subjects.
+  Cursor optional_cursor = 8;
```

Comment:
Should we separate the types of Cursor to show that they can't be mixed and matched between calls that return them. Like should this be a `LookubSubjectCursor`?

Reply:
This is checked by the hashing code, so I think its fine to leave them the same type (which also allows the middleware to read them in a common way, which is important)